src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -281,19 +283,65 @@ private static boolean isDate(String token) {
   }
 
   /**
-   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
-   * it exists, is added to the builder.
+   * Searches the raw text in the builder for the total price using multiple heuristics. If any find
+   * a price, it is added to the builder.
    */
   private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
-    // Assume that the largest price on the receipt is the total price.
+    if (!findPriceAfterTotal(analysisBuilder)) {
+      findLargestPrice(analysisBuilder);
+    }
+  }
+
+  /**
+   * Checks the raw text in the builder for a price appearing after the word "total" and adds it to
+   * the builder if found. The return value indicates whether a price was added.
+   */
+  private static boolean findPriceAfterTotal(AnalysisResults.Builder analysisBuilder) {
+    ImmutableList<String> relevantTokens =
+        getTokensFromRawText(analysisBuilder.getRawText())
+            .filter(token -> isPrice(token) || containsTotal(token))
+            .collect(ImmutableList.toImmutableList());
+
+    double price = 0;
+    boolean priceFound = false;
+    boolean priceFoundAfterMostRecentTotal = true;
+
+    // Try to find the first price after each token containing "total", and keep the last one found.
+    for (String token : relevantTokens) {
+      if (containsTotal(token)) {
+        priceFoundAfterMostRecentTotal = false;
+      } else if (isPrice(token) && !priceFoundAfterMostRecentTotal
```

Comment:
The `isPrice()` check will never be false since the tokens are already filtered so it does not need to be included.